exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -47,21 +47,20 @@ def self.ssl_verify_mode
           end
         end
 
-        def initialize(endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: 'http://localhost:4318/v1/traces'),
+        def initialize(endpoint: nil,
                        certificate_file: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
                        ssl_verify_mode: Exporter.ssl_verify_mode,
                        headers: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS', default: {}),
                        compression: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_COMPRESSION', 'OTEL_EXPORTER_OTLP_COMPRESSION', default: 'gzip'),
                        timeout: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: 10),
                        metrics_reporter: nil)
-          raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" unless OpenTelemetry::Common::Utilities.valid_url?(endpoint)
-          raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || %w[gzip none].include?(compression)
+          @uri = begin
+            prepare_endpoint(endpoint)
+          rescue ArgumentError => e
```

Comment:
what am I missing here? Why is this rescuing an error and then copying the message to a new instance and raising that instead?

Reply:
Ok I see. Did Rubocop fail and tell you that you needed to handle the error raised by `prepare_endpoint`?

Reply:
There's https://github.com/open-telemetry/opentelemetry-ruby/pull/1506#discussion_r1289141806 suggesting to move the code from `prepare_endpoint()` (private method) to the constructor (public) and then throwing the `ArgumentError` from the constructor one as suggested in https://github.com/open-telemetry/opentelemetry-ruby/pull/1506#discussion_r1289028612. However, while pulling in the code from `prepare_endpoint()` to the constructor allows me to throw from the public interface, it also causes cognitive complexity to go up, and then robocop fails.

I'm also quite unhappy with re-throwing the exception and copying the message there, so if you agree, I'd remove it as per your suggestion https://github.com/open-telemetry/opentelemetry-ruby/pull/1506/files#r1331886287. :slightly_smiling_face:

Reply:
Applied [e4d31d6](https://github.com/open-telemetry/opentelemetry-ruby/pull/1506/commits/e4d31d6b58d35ad1f30201df74a08387915f483f) to get rid of it. :slightly_smiling_face: